e2e/validators.go

```diff
@@ -451,7 +451,79 @@ func ValidateContainerd2Properties(ctx context.Context, s *Scenario, versions []
 func ValidateContainerRuntimePlugins(ctx context.Context, s *Scenario) {
 	// nri plugin is enabled by default
 	ValidateDirectoryContent(ctx, s, "/var/run/nri", []string{"nri.sock"})
+}
+
+func ValidateNPDGPUCountPlugin(ctx context.Context, s *Scenario) {
+	command := []string{
+		"set -ex",
+		// Check NPD GPU count plugin config exists
+		"test -f /etc/node-problem-detector.d/custom-plugin-monitor/gpu_checks/custom-plugin-gpu-count.json",
+		// Verify node-problem-detector service is running
+		"systemctl is-active node-problem-detector",
+	}
+	execScriptOnVMForScenarioValidateExitCode(ctx, s, strings.Join(command, "\n"), 0, "NPD GPU count plugin validation failed")
+}
+
+func ValidateNPDGPUCountCondition(ctx context.Context, s *Scenario) {
+	// Wait for NPD to report initial GPU count
+	time.Sleep(30 * time.Second)
+
+	node, err := s.Runtime.Cluster.Kube.Typed.CoreV1().Nodes().Get(ctx, s.Runtime.KubeNodeName, metav1.GetOptions{})
+	require.NoError(s.T, err, "failed to get node %q", s.Runtime.KubeNodeName)
+
+	// Check for GpuCount condition with correct reason
+	var gpuCountCondition *corev1.NodeCondition
+	for i := range node.Status.Conditions {
+		if node.Status.Conditions[i].Type == "GpuCount" && node.Status.Conditions[i].Reason == "GpuCountGood" {
+			gpuCountCondition = &node.Status.Conditions[i]
+			break
+		}
+	}
+
+	require.NotNil(s.T, gpuCountCondition, "expected to find GpuCount condition with GpuCountGood reason on node")
+	require.Equal(s.T, corev1.ConditionFalse, gpuCountCondition.Status, "expected GpuCount condition to be False")
+	require.Contains(s.T, gpuCountCondition.Message, "GPU count is correct", "expected GpuCount message to indicate correct count")
+}
+
+func ValidateNPDGPUCountAfterFailure(ctx context.Context, s *Scenario) {
+	command := []string{
+		"set -ex",
+		// Disable and reset the first GPU
+		"sudo nvidia-smi -i 0 -pm 0", // Disable persistence mode
+		"sudo nvidia-smi -i 0 -c 0",  // Set compute mode to default
+		"PCI_ID=$(sudo nvidia-smi -i 0 --query-gpu=pci.bus_id --format=csv,noheader | sed 's/^0000//')", // sed converts the output into the format needed for NVreg_ExcludeDevices
+		"echo ${PCI_ID} | tee /tmp/npd_test_disabled_pci_id",
+		"echo ${PCI_ID} | sudo tee /sys/bus/pci/devices/${PCI_ID}/driver/unbind", // Reset the GPU
+	}
+	execScriptOnVMForScenarioValidateExitCode(ctx, s, strings.Join(command, "\n"), 0, "failed to disable GPU")
+
+	// Wait for NPD to detect the change
+	time.Sleep(60 * time.Second)
```

Comment:
can we configure the check loop time in NPD with command above to ensure that this sleep is always valid ? (maybe also reduce the sleep time if possible)